pkg/isb/message.go

```diff
@@ -61,6 +61,8 @@ type Header struct {
 	ID string
 	// Key is (key,value) in the map-reduce paradigm which will be used for conditional forwarding.
 	Key string
+	// NumDelivered is the number of times the message has been delivered.
+	NumDelivered uint64
```

Comment:
Considering it's not a major property, shall we put it in something like `metadata`?